src/ShadowDOM/wrappers.js

```diff
@@ -182,7 +182,7 @@ window.ShadowDOMPolyfill = {};
 
   function getDescriptor(source, name) {
     try {
-      return Object.getOwnPropertyDescriptor(source, name);
+      return Object.getOwnPropertyDescriptor(source, name) || dummyDescriptor;
```

Comment:
My expectation is this should throw an exception which we catch in the guard lower down. As the errors happens due to certain conditions with `window` and `showModalDialog`, is there a reason we wouldn't want to check for source === window && name === 'showModalDialog' and only early return the extra dummyDescriptor there?